### DISCUSSION TITLE
Data Importer Security

### DISCUSSION THREAD

**Discussion by vabene1111** [firefly-iii/firefly-iii#5492] [Firefly III] [answered]
Hi, really like the new data importer but i have one question.

I am using the API Token auth for the importer, after first logging into spectre my bank connection is now saved in the importer. This does not seem safe, as even when looking at it from an incognito tab i can still perform an import. 

I haven't seen anything about this in the docs but should I run the importer behind some middleware for security or is the other authentication method better suited for public facing importer setups ?

thanks very much for the great project

**Comment by JC5** [comment, marked as answer]
Yes, you shouldn't link it to the internet. I'll put some extra text in the documentation. But contrary to Firefly III, the data importer doesn't feature extra security measures.

The most secure way would be to add no identifying information (the cloud installation works like this), or only use a token ID and URL.

**Reply by vabene1111** [reply]
Ok thanks, that is a perfectly fine concept I think. I guess I will put some Auth in front of it then.

**Reply by MacPaille** [reply]
On my side I only start my FIDI container when I use it and it's accessible over a domain name that does not exist over internet. My firewall is doing DNS resolver and it only provides access when I4m inside my own network.

Indeed that the tool that needs the most security...
